src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -315,25 +316,25 @@ public void setConnector(HttpConnector connector) {
      * Gets the current rate limit.
      */
     public GHRateLimit getRateLimit() throws IOException {
+        GHRateLimit rateLimit;
         try {
-            return rateLimit = retrieve().to("/rate_limit", JsonRateLimit.class).rate;
+            rateLimit = retrieve().to("/rate_limit", JsonRateLimit.class).rate;
         } catch (FileNotFoundException e) {
             // GitHub Enterprise doesn't have the rate limit, so in that case
             // return some big number that's not too big.
             // see issue #78
-            GHRateLimit r = new GHRateLimit();
-            r.limit = r.remaining = 1000000;
-            long hour = 60L * 60L; // this is madness, storing the date as seconds in a Date object
-            r.reset = new Date(System.currentTimeMillis() / 1000L + hour);
-            return rateLimit = r;
+            rateLimit = GHRateLimit.getPlaceholder();
         }
+
+        return this.rateLimit = rateLimit;
+
     }
 
     /*package*/ void updateRateLimit(@Nonnull GHRateLimit observed) {
         synchronized (headerRateLimitLock) {
             if (headerRateLimit == null
-                    || headerRateLimit.getResetDate().getTime() < observed.getResetDate().getTime()
-                    || headerRateLimit.remaining > observed.remaining) {
+                    || headerRateLimit.getRemaining() > observed.getRemaining()
+                    || headerRateLimit.getResetEpochSeconds() < observed.getResetEpochSeconds()) {
```

Comment:
I must be missing something really simple but... I took a look at the git history and the issues used to discuss the original implementation (back in 2017) and I still don't get it why we wouldn't update the ```headerRateLimit``` variable every time ```updateRateLimit`` method is invoked.

Would you mind sharing with me your understanding of that if statement and/or any idea when this would be applicable?

Reply:
Hm, good question.  Ah, I remember now.  Multi-threading. 
If you have 10 threads all make queries and then all try to update the rate limit.  We want to pick the one with the lowest remaining count, of if a reset occurs we want to pick the one with the later reset date. 

But now that you ask, I that that the check is not quite correct.  I need to write a test that covers this and add comments.

Reply:
ahhhhh, that makes sense  :)